framework/PostProcessors/Metric.py

```diff
@@ -230,7 +263,63 @@ def collectOutput(self,finishedJob, output):
     if isinstance(evaluation, Runners.Error):
       self.raiseAnError(RuntimeError, "Job ", finishedJob.identifier, "failed!")
     outputDict = evaluation[1]
-    output.addRealization(outputDict)
+    # FIXME store the data in DataObjects
+    if isinstance(output, Files.File):
+      availExtens = ['xml']
+      outputExtension = output.getExt().lower()
+      if outputExtension not in availExtens:
+        self.raiseAMessage('Metric postprocessor did not recognize extension ".', str(outputExtension), '". The output will be dumped to a text file')
+      output.setPath(self._workingDir)
+      self.raiseADebug('Write Metric prostprocessor output in file with name: ', output.getAbsFile())
+      output.open('w')
+      self._writeXML(output, outputDict)
+    elif output.type in ['PointSet', 'HistorySet']:
+      self.raiseADebug('Dumping output in data object named', output.name)
+      rlz = {}
+      for key, val in outputDict.items():
+        newKey = key.replace("|","_")
```

Comment:
?

Reply:
This is only the xml output, I just keep the option to output to xml. During the run function, I stored the data as ```metricName|targetVariableName_featureVariableName```, which is easier for me to dump this information into xml. When dumping to the data object, I just replace ```| with _```